File: 프로그래머스/lv2/김미래/요격시스템.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

int solution(vector<vector<int>> targets) {
    int answer = 0;
    int index = 0; 
    sort(targets.begin(), targets.end());
    
    while(index<targets.size()){
        int mark = targets[index][1];
        answer++;
        
        while(index<targets.size() && mark>targets[index][0]){
            if(mark>targets[index][1]){
                mark = targets[index][1];
            }
            index++;
        }
      
    }
    return answer;
}

File: 프로그래머스/lv2/이재표/요격 시스템.java
import java.util.*;
class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        Arrays.sort(targets, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                if (o1[1] != o2[1]) {
                    return o1[1] - o2[1];
                } else {
                    return o1[0] - o2[0];
                }
            }
        });
        int line=0;
        for(int i=0;i<targets.length;i++){
            if(line<=targets[i][0]){
                line=targets[i][1];
                answer++;
            }
        }
        return answer;
    }
}
